### DISCUSSION TITLE
Sequence of responses in MockClient::global

### DISCUSSION THREAD

**Discussion by divdax** [saloonphp/saloon#438] [Q&A]
Hey! 👋

I tried to test a service where I fire multiple `EventRequest`s. In the documentation, I found a sequence of requests, but I need to combine that with other requests besides the `EventRequest`, like so:

```php
$mockClient = Saloon::fake([
  FirstRequest::class => MockResponse::make(['foo' => 'bar']),
  EventRequest::class => [
    MockResponse::make(['event_id' => 1]),
    MockResponse::make(['event_id' => 2]),
  ],
]);
```

Do you maybe know a workaround to test a sequence of requests in this scenario?

**Comment by juse-less** [comment]
@divdax Hey. How about something like this?
```php
Saloon::fake([
    FirstRequest::class => MockResponse::make(['foo' => 'bar']),
    EventRequest::class => static function (PendingRequest $pendindRequest) {
        static $eventId = 1;

        return MockResponse::make(['event_id' => $eventId++]);
    },
])
```

Untested, so it might not work. You may need to move the counter out of the Closure and use it as a reference.

But basically; you can pass a callable as the value so you can handle different requests for the same request class.
If you need access to the request instance being sent, you can access it from `$pendingRequest` in the callable.
